diff --git a/services/clsi/app/js/LatexRunner.js b/services/clsi/app/js/LatexRunner.js
index beefa002ab..fe51b13db2 100644
--- a/services/clsi/app/js/LatexRunner.js
+++ b/services/clsi/app/js/LatexRunner.js
@@ -164,6 +164,10 @@ function _buildLatexCommand(mainFile, opts = {}) {
     '-interaction=batchmode'
   )
 
+  if (process.env.OVERLEAF_ENABLE_SHELL_ESCAPE) {
+    command.push('-shell-escape')
+  }
+
   // Stop on first error option
   if (opts.stopOnFirstError) {
     command.push('-halt-on-error')
